clients/src/test/java/org/apache/kafka/common/serialization/SerializationTest.java

```diff
@@ -84,6 +84,28 @@ public void testStringSerializer() {
     }
 
     @Test
+    public void testShortSerializer() {
+        Short[] shorts = new Short[]{
+            32767,
+            -32768
+        };
+
+        Serializer<Short> serializer = Serdes.Short().serializer();
+        Deserializer<Short> deserializer = Serdes.Short().deserializer();
+
+        for (Short value : shorts) {
+            assertEquals("Should get the original short after serialization and deserialization",
+                    value, deserializer.deserialize(topic, serializer.serialize(topic, value)));
+        }
+
+        assertEquals("Should support null in serialization and deserialization",
```

Comment:
I'd prefer to see this in a separate test, i.e., `shortSerdeShouldSupportNull` or similar